### PR TITLE
allows checkbox widget to set an index in a data tiddler

### PR DESCRIPTION
**Pull request by tobibeer**
- updated documentation
- ~~using a demo macro to actually showcase the wikitext examples~~
    - change wasn't accepted

## Demo

http://2103.tiddlyspot.com

### CODE REVIEW

File: editions/tw5.com/tiddlers/widgets/CheckboxWidget.tid [hunk 1]
```diff
@@ -18,9 +18,10 @@ The content of the `<$checkbox>` widget is displayed within an HTML `<label>` el
 
 |!Attribute |!Description |
 |tiddler |Title of the tiddler to manipulate (defaults to the [[current tiddler|Current Tiddler]]) |
-|tag |The name of the tag to which the checkbox should be bound |
+|tag |The name of the tag to which the checkbox is bound |
 |invertTag |When set to ''yes'', flips the tag binding logic so that the absence of the tag causes the checkbox to be checked  |
-|field |The name of the field to which the checkbox should be bound |
+|field |The name of the field to which the checkbox is bound |
+|index|The index of the //tiddler//, a [[DataTiddler|DataTiddlers]], to which the checkbox is bound<br>» be sure to set the //tiddler// correctly |
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/widgets/CheckboxWidget.tid]: Please tag documentation update with version

**Comment by tobibeer** [comment]
@Jermolene, added "from version" to docs

**Comment by danielo515** [comment]
Holy grail of configuration interfaces. Please merge

File: editions/tw5.com/tiddlers/system/doc-macros.tid [hunk 2]
```diff
@@ -107,3 +107,23 @@ This is an example tiddler. See [[Table-of-Contents Macros (Examples)]].
 </$reveal>
 </$list>
 \end
+
+\define demo(text)
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/system/doc-macros.tid]: Hi @tobibeer were you not able to reuse the macros we've already got in `$:/editions/tw5.com/wikitext-macros`?

File: editions/tw5.com/tiddlers/widgets/CheckboxWidget.tid [hunk 3]
```diff
@@ -18,9 +18,10 @@ The content of the `<$checkbox>` widget is displayed within an HTML `<label>` el
 
 |!Attribute |!Description |
 |tiddler |Title of the tiddler to manipulate (defaults to the [[current tiddler|Current Tiddler]]) |
-|tag |The name of the tag to which the checkbox should be bound |
+|tag |The name of the tag to which the checkbox is bound |
 |invertTag |When set to ''yes'', flips the tag binding logic so that the absence of the tag causes the checkbox to be checked  |
-|field |The name of the field to which the checkbox should be bound |
+|field |The name of the field to which the checkbox is bound |
+|index|<<.from-version "5.1.14">> The index of the //tiddler//, a [[DataTiddler|DataTiddlers]], to which the checkbox is bound<br>» be sure to set the //tiddler// correctly |
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/widgets/CheckboxWidget.tid]: The use of `<br>»` is a new convention that isn't used anywhere else in the documentation. To match the existing style, perhaps use the `.tip` macro after the table?

File: editions/tw5.com/tiddlers/system/doc-macros.tid (hunk 2 above)
Comment (tobibeer) [inline editions/tw5.com/tiddlers/system/doc-macros.tid]: I can and I guess me not liking neither the unnecessarily wordy names of those, nor the presentation does warrant proposing a different macro name or presentation in some PR.

File: editions/tw5.com/tiddlers/widgets/CheckboxWidget.tid (hunk 3 above)
Comment (tobibeer) [inline editions/tw5.com/tiddlers/widgets/CheckboxWidget.tid]: Haven't paid much attention to documentation macros. Using the .tip macro looks good enough to me.

**Comment by pmario** [comment]
@tobibeer .. It seems this PR changed 167 files ?! .. There's something wrong.

**Comment by tobibeer** [comment]
@pmario, no idea why github thinks that. If you look at the files it says "nothing changed".

**Comment by Jermolene** [comment]
Hi @tobibeer it looks like you've accidentally changed the file mode of a bunch of core files. Have a look at the "Files changed" tab here on GitHub:

screenshot 2016-12-17 11 48 42

**Comment by pmario** [comment]
` boot/sjcl-license.txt 100755 → 100644 `
yea, but it seems the access rights have been changed. ... I'll need to check in on my system. 

which OS dou you use and which git program?

**Comment by pmario** [comment]
@Jermolene .. That's funny. On my master branch they are already 644 .. what's about your system?

**Comment by Jermolene** [comment]
Hi @tobibeer indeed, I can confirm that in my copy of the repo there are a bunch of files with wrong file modes, corresponding to the files that are being corrected here.

I wonder if you can somehow persuade your set-up to create a pull request that just has the corrections to the file modes? You need to make sure that you don't accidentally reset the executable modes from the scripts in /bin.

**Comment by tobibeer** [comment]
squashed those commits.

**Comment by Jermolene** [comment]
Thanks @tobibeer -- I'd still like to fix the file modes if you're able to do a PR.

**Comment by tobibeer** [comment]
I am on windows.

**Comment by tobibeer** [comment]
No idea what those file-modes are and how to fix anything about them.

**Comment by Jermolene** [comment]
> I am on windows.

Right, but somehow your set-up included those corrections in the PR that you created. So I'm guessing there must have been some part of the process that reset the file modes, and included it in the PR?

**Comment by tobibeer** [comment]
I'm guessing the desktop client somehow can run into race conditions with the console or whichever processes it is made up of compete over who gets to decide what to do next. In other words, the UI sometimes doesn't appear to reflect what I just did on the console ...and there's no way to just refresh the UI. Even restarting it didn't help. anyhow, I guess squashing did the trick.

**Pull request merged**